scalafix-rules/src/main/scala/scalafix/internal/pc/Embedded.scala

```diff
@@ -0,0 +1,106 @@
+package scalafix.internal.pc
+
+import java.net.URLClassLoader
+import java.nio.file.Path
+import java.util.ServiceLoader
+
+import scala.jdk.CollectionConverters._
+
+import scala.meta.pc.PresentationCompiler
+
+import coursierapi.Dependency
+import coursierapi.Fetch
+import coursierapi.MavenRepository
+
+object Embedded {
+
+  def presentationCompiler(
+      scalaVersion: String
+  ): PresentationCompiler = {
+    val deps =
+      scala3PresentationCompilerDependencies(scalaVersion)
+    val jars = Fetch
+      .create()
+      .addDependencies(deps: _*)
+      .addRepositories(
+        MavenRepository.of(
+          "https://oss.sonatype.org/content/repositories/snapshots"
+        )
+      )
```

Comment:
This might not be enough in corporate/CI environments. Custom repositories provided at Scalafix invocation are not available to rules, so I am not sure how to honor them without changing the API.

Reply:
Setting COURSIER_REPOSITORIES globally should help, but it's always a bit of a problem. The other options would be to somehow reuse sbt resolution? And just bring in the exact presentation compiler version via that route?

Reply:
That might be possible, and it would solve it for sbt-scalafix, but not for other clients. Anyway, it's not a blocker for this PR.

Reply:
I guess we could just depend on the presentation compiler and override compiler version in sbt (metals should be able to do that also I think). For all other clients we could just publish LTS/Next which in most cases should be enough.